src/views/Home/components/DaysLeft/index.jsx

```diff
@@ -24,65 +33,145 @@ const DaysLeft = () => {
       setFirstDay(first);
       setLastDay(last);
       setCurrentSessionDescription(currentSessionDescription);
       setDaysRemaining(daysRemaining);
       setDaysFinished(daysInSemester - daysRemaining || 0);
       setLoading(false);
+      setFallEnd(fallEnd);
+      setSpringBegin(springBegin);
+      setSummerBegin(summerBegin);
+      setSummerEnd(summerEnd);
+      setSpringEnd(springEnd);
+      setFallBegin(fallBegin);
+      setNextFallBegin(nextFallBegin);
     };
 
     load();
   }, []);
 
-  const colors = theme360.colorSchemes.light.palette;
+  const academicTerms = ['Fall', 'Spring', 'Summer'];
+
+  const termBreaks = ['Winter', 'PreSummer', 'PostSummer'];
+
+  const msToDays = 1.15741e-8;
+  const winterBreak = Math.round((springBegin - fallEnd) * msToDays);
+  const daysRemainingWinter = Math.round((springBegin - today) * msToDays);
+  const winterProgress = (Math.round((today - fallEnd) * msToDays) / winterBreak) * 100;
+  const preSummer = Math.round((summerBegin - springEnd) * msToDays);
+  const preSummerProgress = (Math.round((today - springEnd) * msToDays) / preSummer) * 100;
+  const daysRemainingPreSummer = Math.round((summerBegin - today) * msToDays);
+  const postSummer = Math.round((nextFallBegin - summerEnd) * msToDays);
+  const daysRemainingPostSummer = Math.round((nextFallBegin - today) * msToDays);
+  const postSummerProgress = (Math.round((today - summerEnd) * msToDays) / postSummer) * 100;
+
+  const termLoop = [
+    { start: fallBegin, end: fallEnd, type: academicTerms[0] },
+    { start: fallEnd, end: springBegin, type: termBreaks[0] },
+    { start: springBegin, end: springEnd, type: academicTerms[1] },
+    { start: springEnd, end: summerBegin, type: termBreaks[1] },
+    { start: summerBegin, end: summerEnd, type: academicTerms[2] },
+    { start: summerEnd, end: nextFallBegin, type: termBreaks[2], condition: postSummer > 1 },
+  ];
+
+  let termFinder = academicTerms[0];
+
+  for (let term of termLoop) {
+    if (
+      today >= term.start &&
+      today <= term.end &&
+      (term.condition === undefined || term.condition)
+    ) {
+      termFinder = term.type;
+      break;
+    }
+  }
+
+  const daysRemainingTermBreaks =
+    termFinder === termBreaks[0]
+      ? daysRemainingWinter
+      : termFinder === termBreaks[1]
+        ? daysRemainingPreSummer
+        : termFinder === termBreaks[2]
+          ? daysRemainingPostSummer
+          : '';
+
+  const termLabelLoop = [
+    { term: academicTerms[0], label: academicTerms[0] },
+    { term: academicTerms[1], label: academicTerms[1] },
+    { term: academicTerms[2], label: academicTerms[2] },
+    { term: termBreaks[0], label: academicTerms[1] },
+    { term: termBreaks[1], label: academicTerms[2] },
+    { term: termBreaks[2], label: academicTerms[0] },
+  ];
+
+  let termLabel = academicTerms[0];
+
+  for (let currentLabel of termLabelLoop) {
+    if (termFinder === currentLabel.term) {
+      termLabel = currentLabel.label;
+    }
+  }
```

Comment:
same as above. You can put these in functions that are called in a useEffect to prevent recalculations each time the page is loaded.